### PR TITLE
Update KZ holidays: add 2025 special holidays

### PR DESCRIPTION
**Pull request by qaziqarta**
## Proposed change

Update KZ holidays: add 2025 special and substituted holidays.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new `holidays` functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've successfully run `make check`, all checks and tests are green

[contributing-guidelines]: https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/holidays/tree/dev/docs/source

### CODE REVIEW

File: holidays/countries/kazakhstan.py [hunk 1]
```diff
@@ -195,6 +196,7 @@ class KazakhstanIslamicHolidays(_CustomIslamicHolidays):
         2022: (JUL, 9),
         2023: (JUN, 28),
         2024: (JUN, 16),
+        2025: (JUN, 6),
```
Comment (PPsyrius) [inline holidays/countries/kazakhstan.py:199]: You might still want to add the source for Kazakhstan's 2025 Kurban Ait/Eid al-Adha date though i.e. https://en.tengrinews.kz/kazakhstan_news/ramadan-2025-start-date-announced-for-kazakhstan-264556/

**Review by PPsyrius — DISMISSED** [review]
LGTM 🇰🇿

File: holidays/countries/kazakhstan.py [hunk 2]
```diff
@@ -74,6 +74,10 @@ class Kazakhstan(
         - `2022 <https://adilet.zan.kz/kaz/docs/P2200000796>`_
         - `2023 <https://adilet.zan.kz/kaz/docs/P2300000326>`_
         - `2024 <https://adilet.zan.kz/kaz/docs/G24G0000109>`_
+        - `2025 <https://adilet.zan.kz/rus/docs/G24G0000436>`_
```
Comment (KJhellico) [inline holidays/countries/kazakhstan.py:77]:
```suggestion
        - `2025 <https://adilet.zan.kz/kaz/docs/G24G0000436>`_
```
For consistency with previous years.

**Review by PPsyrius — APPROVED** [review]

**Review by KJhellico — APPROVED** [review]
LGTM.

**Pull request merged**